lib/api/src/errors.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{LinkError, RuntimeError};
+use thiserror::Error;
+
+// /// An error while instantiating a module.
+// ///
+// /// This is not a common WebAssembly error, however
```

Comment:
Need to standarize errors in wasmer_types

Reply:
I standardize everything except the `InstantiationError` as that error depends on the `RuntimeError` which is different for sys and js (since the sys one includes a backtrace that can't be used on js)